### ISSUE TITLE
Use boto3's default behaviour to decide the default region

### ISSUE THREAD

**Issue by rubendura** [fugue/credstash#93]
Right now credstash uses the environment variable AWS_DEFAULT_REGION or the -r CLI parameter to specify in which region credstash should operate. If none of those are available it defaults forcibly to us-east-1 (this is hardcoded).

Boto3 will use one of the configured AWS profiles, for which a default region can be specified. Boto3 will use that region as the default when creating any client. Plus, Boto3 will use as a default the EC2 instance region when using Instance Profiles to fetch credentials.

Is there any reason to have us-east-1 hardcoded instead of relying in the default behaviour of boto3? I've already fallen trap of this a couple time by forgetting to specify a region and not realising why it was not working.

**Comment by artburkart** [comment]
I've opened a PR to tackle this problem. I don't think it can really be done without breaking backwards compatibility though... not unless we added a new arg to indicate that we want to use the default AWS credential chain. Not sure if that's worth it though.

**Comment by wayne-luminal** [comment]
I agree. I'd like to remove the hardcoded default but keep the `-r` parameter in the next major version. There is some discussion in #180 if you want to comment there as well.
